tests/components/homematicip_cloud/test_hap.py

```diff
@@ -132,21 +111,17 @@
     hass = Mock()
     entry = Mock()
     home = Mock()
-    home.disable_events = mock_coro_func()
-    entry.data = {
-        hmipc.HMIPC_HAPID: "ABC123",
-        hmipc.HMIPC_AUTHTOKEN: "123",
-        hmipc.HMIPC_NAME: "hmip",
-    }
-    hap = hmipc.HomematicipHAP(hass, entry)
-    with patch.object(hap, "get_hap", return_value=mock_coro(home)):
+    home.disable_events = CoroutineMock()
+    entry.data = {HMIPC_HAPID: "ABC123", HMIPC_AUTHTOKEN: "123", HMIPC_NAME: "hmip"}
+    hap = HomematicipHAP(hass, entry)
+    with patch.object(hap, "get_hap", return_value=home):
         assert await hap.async_setup()
 
     assert hap.home is home
     assert not hass.services.async_register.mock_calls
     assert len(hass.config_entries.async_forward_entry_setup.mock_calls) == 8
 
-    hass.config_entries.async_forward_entry_unload.return_value = mock_coro(True)
+    hass.config_entries.async_forward_entry_unload = CoroutineMock(return_value=True)
```

Comment:
We should avoid patching the core. It's better to keep it intact and assert the state of the core, eg that the entry has been unloaded, `ENTRY_STATE_NOT_LOADED`.

https://github.com/home-assistant/home-assistant/blob/1b3c4ed4b3250e4c8fcf6bebab4c36fc5bf0da7c/homeassistant/config_entries.py#L156-L157

Reply:
OK. I completely rewrote the test to:
- avoid patching the core
- cover method hap.async_reset()
- show the unloading of the entry

Hope that fits better.